test/test_stablehlo_dump.py

```diff
@@ -0,0 +1,27 @@
+import torch_xla
+import torch_xla.core.xla_model as xm
+import torch
+'''
+The following MLIR module should be dump in this test
+module @IrToHlo.9 attributes {mhlo.cross_program_prefetches = [], mhlo.dynamic_parameter_bindings = [], mhlo.is_dynamic = false, mhlo.use_auto_spmd_partitioning = false} {
+  func.func @main(%arg0: tensor<1xi64>, %arg1: tensor<1xi64>) -> tuple<tensor<1xi64>, tensor<1xi64>, tensor<1xi64>> {
+    %0 = stablehlo.constant dense<1> : tensor<i64>
+    %1 = stablehlo.constant dense<1> : tensor<1xi64>
+    %2 = stablehlo.multiply %arg1, %1 : tensor<1xi64>
+    %3 = stablehlo.add %arg0, %2 : tensor<1xi64>
+    %4 = stablehlo.tuple %arg0, %arg1, %3 {xla_shape = "(s64[1]{0}, s64[1]{0}, s64[1]{0})"} : tuple<tensor<1xi64>, tensor<1xi64>, tensor<1xi64>>
+    return %4 : tuple<tensor<1xi64>, tensor<1xi64>, tensor<1xi64>>
+  }
+}
+'''
+
+x = torch.tensor([3], device=xm.xla_device())
+y = torch.tensor([3], device=xm.xla_device())
+z = x + y
+
+# Example usage of dumping StableHLO given output tensors
+stablehlo = xm.get_xla_tensors_stablehlostep([z])
+print(stablehlo)
+# Example usage of dump StableHLO of the entire graph
+stablehlo = xm.xla_get_stablehlo()
```

Comment:
Maybe we don't need 2 api, we just need one api that takes optional arguments. If no argument tensor is passed, we dump the whole graph.

Reply:
Yes, that's a good idea. Let's keep the `xla_get_stablehlo`

Reply:
good ideas here - at this stage, based on our discussions today, we need to test:

1- dumping stablehlo string (with optional tensor input)
2- converting to stablehlo and doing the actual lowering

I don't think this test addresses the second bullet

Reply:
This PR focuses on StableHLO dump API - see description.

For 2, did you mean compiling & compilation with StableHLO?

Reply:
Possibly related (maybe more related to 2), is there any way to run a test suite and either:
1. Dump StableHLO for each test
2. Convert to StableHLO and compile

An environment variable or otherwise? This will be useful as we add support for the remaining pieces needed by PyTorch/XLA in StableHLO, to make sure we have everything covered.

Reply:
That's a good idea👍 

1. The environment variables to enable StableHLO dump is added in the PR. Details in the "Enable StableHLO dump via Debugging flag" section in the description. However, we don't have a workflow to validate the dump from the tests. Maybe adding a separate Python script that compares the dump with the pre-generated dump is a way to go.

2. This is not covered by this PR. Will include this workflow in the 2nd PR that changes the compilation workflow. This PR focuses on the StableHLO dump.